include/zephyr/drivers/sent/sent.h

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Single Edge Nibble Transmission (SENT) driver API.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SENT_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SENT_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief SENT Interface
+ * @defgroup sent_interface SENT Interface
+ * @since 4.2
+ * @version 0.1.0
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @brief SENT frame type
+ */
+enum sent_frame_type {
+	SENT_SHORT_SERIAL_FRAME,
+	SENT_ENHANCED_SERIAL_FRAME_4_BIT_ID,
+	SENT_ENHANCED_SERIAL_FRAME_8_BIT_ID,
+	SENT_FAST_FRAME
+};
+
+/**
+ * @brief Maximum number of data nibbles
+ */
+#define SENT_MAX_DATA_NUBBLES 8
+
+/**
+ * @brief SENT frame structure
+ */
+struct sent_frame {
+	enum sent_frame_type type;
+
+	union {
+		struct {
+			uint8_t id;
+			uint16_t data;
+		} serial;
+
+		struct {
+			uint8_t data_nibbles[SENT_MAX_DATA_NUBBLES];
```

Comment:
I have changed the SENT fast frame structure to use an array of data nibbles instead of a single 32-bit data value.
And updated shim-driver and test, sample code to work with this change.